src/react/AppProvider.jsx

```diff
@@ -61,8 +62,28 @@ export default function AppProvider({ store, children }) {
     setLocale(getLocale());
   });
 
+  useEffect(() => {
+    if (config.THEME_OVERRIDE_URL) {
+      const themeLink = document.createElement('link');
+      themeLink.href = config.THEME_OVERRIDE_URL;
+      themeLink.rel = 'stylesheet';
+      themeLink.type = 'text/css';
+      themeLink.onload = () => setThemeLoaded(true);
+      themeLink.onerror = () => setThemeLoaded(true);
+
+      document.head.appendChild(themeLink);
+
+      return () => document.head.removeChild(themeLink);
+    }
+    setThemeLoaded(true);
+  }, [config.THEME_OVERRIDE_URL]);
+
   const appContextValue = useMemo(() => ({ authenticatedUser, config, locale }), [authenticatedUser, config, locale]);
 
+  if (!themeLoaded) {
```

Comment:
don't show content until theme override loaded to avoid showing content with default theme, before the overrides have been applied